platform/mbed_toolchain.h

```diff
@@ -514,6 +514,13 @@ typedef int FILEHANDLE;
 #define MBED_NONSECURE_ENTRY
 #endif
 
+/* The code for the main() function must be emitted even if it appears that
+ * the function is not referenced.
+ * This is the case for combined "-Wl,--wrap,main" and "-flto" flags with
+ * the GCC_ARM toolchain.
+ */
+MBED_USED int main(void);
```

Comment:
This is causing errors in the build , please review latest test logs

Reply:
PS - have you checked that this is still needed with GCC 9? You added it during GCC 8 testing.

Reply:
Yes, GCC9 still needs a `used` attribute for wrapped main(). I'd like to move this to compiler/linker flags but so far I've only found `-fkeep-static-functions` which wastes a lot of lto gains (58408 RAM & 384236 Flash for the release profile).

Reply:
Removing `"-Wl,--wrap,main"` also solves the issue. But I suppose I'm not allowed to touch that. https://github.com/ARMmbed/mbed-os/blob/7fbe0c6b416033a4a89ef07fe9a63990de83242c/rtos/source/TARGET_CORTEX/TOOLCHAIN_GCC_ARM/mbed_boot_gcc_arm.c#L73-L78

Reply:
we would loose `mbed_main` , wouldn't we ?
 
is wrap actually working fine with lto (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88643) in GCC ARM 9?